src/Compilers/CSharp/Test/Emit/CodeGen/StackAllocInitializerTests.cs

```diff
@@ -0,0 +1,388 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.StackAllocInitializer)]
+    public class StackAllocInitializerTests : CompilingTestBase
+    {
+        [Fact]
+        public void TestUnused()
+        {
+            var text = @"
+using System;
+
+static unsafe class C
+{
+    static byte Method()
+    {
+        return 42;
+    }
+
+    static void Main()
+    {
+        byte* p = stackalloc byte[3] { 42, Method(), 42 };
+    }
+}
+";
+            CompileAndVerify(text,
```

Comment:
Please also verify execution order with `expectedOutput:` and printing inside `Method` and some other method.